app/services/pagarme/api.rb

```diff
@@ -1,18 +1,38 @@
 class Pagarme::API
+  class ConnectionError       < StandardError; end
+  class ResourceNotFound      < StandardError; end
+  class InvalidAttributeError < StandardError; end
+
   class << self
     def find_plan(plan_id)
       PagarMe::Plan.find_by_id(plan_id)
+    rescue PagarMe::ConnectionError
+      raise_connection_error
```

Comment:
Não acho que é necessário um método nesse caso aqui, pode usar só : 
raise ConnectionError, 'The connection with our payment server was lost'